testrail/api.py

```diff
@@ -257,6 +263,29 @@ def run_with_id(self, run_id):
         except IndexError:
             raise TestRailError("Run ID '%s' was not found" % run_id)
 
+    @UpdateCache(_shared_state['_runs'])
+    def add_run(self, run):
+        fields = ['name', 'description', 'suite_id', 'milestone_id',
+                  'assignedto_id', 'include_all', 'case_ids']
+        project_id = run.get('project_id')
+        payload = self._payload_gen(fields, run)
+        return self._post('add_run/%s' % project_id, payload)
+
+    @UpdateCache(_shared_state['_runs'])
```

Comment:
I am a little worried about performance impacts with reseting cache for every post.  I think we are forced to for creating a new run because we need to get the id.  But when updating and deleting I think we should just modify the cache after a successful post.  This way the cache stays current but doesn't require a new get request.